doc/user/content/sql/subscribe.md

```diff
@@ -288,6 +291,121 @@ If your row has a unique column key, it is possible to map the update to its cor
 
 In the example above, `Column 1` acts as the column key that uniquely identifies the origin row the update refers to; in case this was unknown, hashing the values from `Column 1` to `Column N` would identify the origin row.
 
+[//]: # "TODO(morsapaes) This page is now complex enough that we should
+restructure it using the same feature-oriented approach as CREATE SOURCE/SINK.
+See #18829 for the design doc."
+
+### Modifying the output format
+
+{{< alpha />}}
+
+#### `ENVELOPE UPSERT`
+
+To modify the output of `SUBSCRIBE` to support upserts, use `ENVELOPE UPSERT`.
+This clause allows you to specify a `KEY` that Materialize uses to interpret
+the rows as a series of inserts, updates and deletes within each distinct
+timestamp.
+
+* Using this modifier, the output rows will have the following
+structure:
+
+   ```sql
+   SUBSCRIBE mview ENVELOPE UPSERT (KEY (key));
+   ```
+
+   ```sql
+   mz_timestamp | mz_state | key  | value
+   -------------|----------|------|--------
+   100          | upsert   | 1    | 2
```

Comment:
I think we need to document that using `UPSERT` causes the order of the columns to change as compared to a normal `SUBSCRIBE`, with the columns of the key outputted first.

Reply:
good catch, thanks

Reply:
Ah, Andrew originally included this! I cut it since it should be obvious from the example, but in hindsight it's a pretty relevant detail for consumer applications.